sisyphus/hash.py

```diff
@@ -61,6 +62,10 @@ def get_object_state(obj):
         assert args is not None, "Failed to get object state of: %s" % repr(obj)
         state = None
 
+    if isinstance(obj, enum.Enum):
+        assert isinstance(state, dict)
+        state.pop("_sort_order_", None)  # compat with Python <=3.10, https://github.com/rwth-i6/sisyphus/issues/188
```

Comment:
compat <= 3.10 could also mean that this would crash otherwise. 
It would be good to make clear that is done to keep the computed hash equal with python <=3.10.